File: leetcode/2232-adding-spaces-to-a-string/solution.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        string ret;
        int n = s.size(), m = spaces.size();
        ret.reserve(n+m);
        int i = 0;
        for(int j: spaces) {
            ret.append(s, i, j-i);
            ret.push_back(' ');
            i = j;
        }
        ret.append(s, i, n);
        return ret;
    }
};

File: leetcode/2232-adding-spaces-to-a-string/solution.rs
impl Solution {
    pub fn add_spaces(s: String, spaces: Vec<i32>) -> String {
        let mut ret = String::with_capacity(s.len()+spaces.len());
        let mut i = 0;
        for j in spaces {
            ret.push_str(&s[i..j as usize]);
            ret.push(' ');
            i = j as usize;
        }
        ret.push_str(&s[i..s.len()]);
        ret
    }
}
